### ISSUE TITLE
Allow users to add / edit / remove segments from binary

### ISSUE THREAD

**Issue by uxmal** [uxmal/reko#134]
Issue #122 uncovered the need for users to be able to specify segments of valid address space that haven't been specified by the binary. One instance is from #122 itself, where the binary is referencing global addresses in a range not covered by the binary itself. Another example is an MS-DOS binary that after being loaded accesses segment 0A000h, where the VGA adapter memory resides.

Uses should be allowed to add a segment to a binary by right-clicking it in the browser and selecting `Add segment` from the menu. Adding a segment requires supplying a name (could be auto generated) and an address range (either `address1 address2` or `address1 L length-in-bytes`. Segment addresses and extents should be modifiable (remember, the user is the Oracle and Knows What He/She Is Doing). Finally segments should be removeable.

This has consequences for creation of ImageReaders and the topic of #78 which deals with how Reko should treat the program image -- right now it assumes it is a long block of bytes, but that can't work if you allow very sparse binaries.

**Comment by lab313ru** [comment]
Yep, that m68k code was cut from Sega Genesis ROM. There are many memory segments there: http://www.db-elec.com/home/technical-info/Genesis/memory-map

**Comment by uxmal** [comment]
Nice reference! I will add the memory map to the `SegaGenesisPlatform`. That will help a lot.

**Comment by lab313ru** [comment]
Ask me, if something is not so clear there.
